index.js

```diff
@@ -112,9 +144,19 @@ FastForEach.prototype.onArrayChange = function (changeSet) {
     added: [],
     deleted: [],
   };
-  ko.utils.arrayForEach(changeSet, function(changeItem) {
-    changeMap[changeItem.status].push(changeItem);
-  });
+  for (var i = 0, len = changeSet.length; i < len; i++) {
+    // the change is appended to a last change info object when both are 'added' and have indexes next to each other
+    // here I presume that ko is sending changes in monotonic order (in index variable) which happens to be true, tested with push and splice with multiple pushed values
+    if (i > 0 && changeSet[i].status === "added" && changeSet[i - 1].status === "added" && changeSet[i - 1].index === changeSet[i].index - 1) {
```

Comment:
Much cleaner, thanks! 

I will try to mull some tests for this today, and merge soon. Will report back here.

Thanks, @cervengoc